v0/pipeline/steps/cd/k8s-istio-provider-spec.yaml

```diff
@@ -0,0 +1,25 @@
+title: IstioProviderSpec
+type: object
+required:
+  - gateways
```

Comment:
we should make gateways and hosts fields not mandatory

Reply:
should we update tech spec in that case? 
<img width="880" alt="Screenshot 2023-11-10 at 12 21 19" src="https://github.com/harness/harness-schema/assets/98389528/c4eb2c28-f440-4cef-9007-fc9ebdaf1d2d">

Reply:
done